cypress/integration/test_table.js

```diff
@@ -95,4 +95,86 @@ describe("tests table", function() {
       "_blank"
     );
   });
+
+  describe("Test Status Selector", () => {
+    beforeEach(() => {
+      cy.visit(TESTS_ROUTE);
+      cy.get("#cy-test-status-select > .cy-treeselect-bar").click();
+    });
+
+    it("Status select says 'No filters selected' by default", () => {
+      cy.get("#cy-test-status-select").contains("No filters selected");
+    });
+
+    it("Clicking on 'All' checkbox adds all statuses to URL ", () => {
+      cy.get(".cy-checkbox")
+        .contains("All")
+        .click();
+      cy.location().should(loc => {
+        expect(loc.pathname).to.equal(TESTS_ROUTE);
+        expect(loc.search).to.include("statuses=all,pass,fail,skip,silentfail");
+      });
+    });
+
+    const statuses = [
+      { display: "Pass", key: "pass" },
+      { display: "Silent Fail", key: "silentfail" },
+      { display: "Fail", key: "fail" },
+      { display: "Skip", key: "skip" }
+    ];
+
+    statuses.forEach(({ display, key }) => {
+      it(`Clicking on ${display} status checkbox adds ${key} status to URL and clicking again removes it`, () => {
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+        cy.location().should(loc => {
+          expect(loc.pathname).to.equal(TESTS_ROUTE);
+          expect(loc.search).to.include(`statuses=${key}`);
+        });
+        cy.wait(100);
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+        cy.location().should(loc => {
+          expect(loc.pathname).to.equal(TESTS_ROUTE);
+          expect(loc.search).to.not.include(`statuses=${key}`);
+        });
+      });
+    });
+
+    it("Checking all statuses will check 'All' checkbox", () => {
+      statuses.forEach(({ display }) => {
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+      });
+      cy.get(".cy-checkbox > input").should("be.checked");
+    });
+
+    it("Unchecking a status will uncheck 'All' checkbox", () => {
+      statuses.forEach(({ display }) => {
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+      });
+      cy.get(".cy-checkbox")
+        .contains("Pass")
+        .click();
+      cy.get(".cy-checkbox > input")
+        .first()
+        .should("not.be.checked");
+    });
+
+    it("Checking multiple statuses adds them to URL", () => {
+      statuses.forEach(({ display }) => {
+        cy.get(".cy-checkbox")
+          .contains(display)
+          .click();
+      });
+      cy.location().should(loc => {
+        expect(loc.search).to.include("statuses=pass,silentfail,fail,skip,all");
+      });
+    });
+  });
```

Comment:
i'm not sure what this test is doing

Reply:
This tests shows that clicking on multiple statuses adds them all to the URL as opposed to one, some or none

Reply:
what do you think of updating the name of the test to say exactly that? because that i understand :)